### PR TITLE
memeber list (for gameserver ini feeding)

### PR DESCRIPTION
**Pull request by swomma**
For 29th/personnel#231
Update to make scripts updating squad tags in gameservers return correct unit (used to return LH, Adj, etc).
In L:82 if you set `ORDER BY` to `class ASC, path ASC` you get a top combat assignment, if set to `class ASC, path DESC`  you get a lowest combat assignment

### CODE REVIEW

File: application/controllers/assignments.php [hunk 1]
```diff
@@ -42,14 +42,20 @@ private function calculate_duration($assignments, $member_id) {
         $this->discharge_model->order_by('date DESC');
         $gdDate = $this->discharge_model->get()->result_array();
         $gdDate = ( sizeof($gdDate) ? $gdDate[0]['date'] : null );
-        foreach($assignments as $assignment) {
+        foreach($assignments as $assignment) 
+        {
+          if ( !($assignment['unit']['class'] == 'Training') )
```
Comment (wilson29thid) [inline application/controllers/assignments.php]: Why was this added? In the past, service duration started at enlistment date, not graduation date.

File: application/controllers/qualifications.php [hunk 2]
```diff
@@ -15,18 +15,21 @@ public function view_options() { $this->response(array('status' => true)); }
...
-            if($member_id) {
+            if($unit_id) {
+              $model->by_unit($unit_id);  
+            }
+            elseif($member_id) {
                 $model->where('qualifications.member_id', $member_id);
             }
             $qualifications = nest($model->get()->result_array());
-            $this->response(array('status' => true, 'qualifications' => $qualifications));
+            $this->response(array('status' => true, 'qualifications' => $qualifications, 'mem' => $member_id, 'unit' => $unit_id ));
```
Comment (wilson29thid) [inline application/controllers/qualifications.php]: Why are these included in the response? This isn't consistent with the other collections, is it?

File: application/controllers/units.php [hunk 3]
```diff
@@ -214,6 +214,35 @@ public function discharges_get($filter) {
     }
     
     /**
+     * UNIT QUALIFICATIONS
```
Comment (wilson29thid) [inline application/controllers/units.php]: Any reason this controller can't simply inherit this method from `MY_Controller.php` like the other similar controllers?

File: application/models/discharge_model.php [hunk 4]
```diff
@@ -78,7 +78,7 @@ public function default_order_by() {
      * Should match MY_Model's method except filtering by active assignments
      */ 
     public function by_unit($unit_id) {
-        $this->filter_join('assignments', 'assignments.member_id = ' . $this->table . '.member_id', 'left');
+        $this->filter_join('assignments', 'assignments.member_id = ' . $this->table . '.member_id AND (assignments.end_date=discharges.date)', 'left');
```
Comment (wilson29thid) [inline application/models/discharge_model.php]: Can you explain why this was added?

File: application/models/member_model.php [hunk 5]
```diff
@@ -79,7 +79,7 @@ public function default_join() {
             //->join('assignments', 'assignments.id = members.primary_assignment_id', 'left')
             ->join('assignments', 'assignments.member_id = members.id AND (assignments.start_date <= CURDATE()) AND (assignments.end_date > CURDATE() OR assignments.end_date IS NULL)', 'left')
             ->join('positions', 'positions.id = assignments.position_id', 'left')
-            ->join('units', 'units.id = assignments.unit_id', 'left')
+            ->join('(SELECT * FROM units ORDER BY class ASC, path) AS units', 'units.id = assignments.unit_id','left')
```
Comment (wilson29thid) [inline application/models/member_model.php]: Huh? What's the difference here? Order wouldn't matter because of the join conditional following it, would it?

File: application/models/discharge_model.php (hunk 4 above)
Comment (swomma) [inline application/models/discharge_model.php]: To prevent old members' discharges (members that were transfered out of the unit prior to discharge) showing in unit activity feed

File: application/models/qualification_model.php [hunk 6]
```diff
@@ -54,4 +54,25 @@ public function select_member() {
     public function order_by() {
         $this->db->order_by('qualifications.date DESC');
     }
+
+    public function by_unit($unit_id) {
```
Comment (wilson29thid) [inline application/models/qualification_model.php]: Do you know if this can simply be inherited from `MY_Model.php`?

File: application/controllers/assignments.php (hunk 1 above)
Comment (swomma) [inline application/controllers/assignments.php]: It was changed as ordered by major @Wheatles29 on 23 Feb

File: application/controllers/qualifications.php (hunk 2 above)
Comment (swomma) [inline application/controllers/qualifications.php]: How else would you get qualifications for the unit and its subunits from last X (where X=30 for now) days?

File: application/models/member_model.php (hunk 5 above)
Comment (swomma) [inline application/models/member_model.php]: Of course it does matter as we only include in API response only the first accurence of member - and that would include the first unit assigned to - otherwise it was by date, descending - it is made so to prevent Staff units showing in DH / RS scoreboard

File: application/models/qualification_model.php (hunk 6 above)
Comment (swomma) [inline application/models/qualification_model.php]: I have no idea - I tried to do it as it was done in other models
